Mypy Errors with new version of mypy

## 🐛 Bug description
With the new version of mypy, ignite/handlers/checkpoint.py and ignite/utils.py has some errors related to typing.
```
ignite/utils.py:161: error: unused "type: ignore" comment
        existing = name is None or name in logging.root.manager.loggerDict...
        ^
ignite/handlers/checkpoint.py:9: error: unused "type: ignore" comment
    from tempfile import _TemporaryFileWrapper  # type: ignore[attr-define...
    ^
ignite/handlers/checkpoint.py:695: error: Incompatible types in assignment
(expression has type "None", variable has type "_TemporaryFileWrapper") 
[assignment]
                tmp = None  # type: _TemporaryFileWrapper
                      ^
ignite/handlers/checkpoint.py:697: error: Incompatible types in assignment
(expression has type "IO[bytes]", variable has type "_TemporaryFileWrapper") 
[assignment]
                    tmp = tempfile.NamedTemporaryFile(delete=False, dir=se...
                          ^
Found 4 errors in 2 files (checked 98 source files)
```


## Environment

- Ignite Version (e.g., 0.3.0): Current master branch
- OS (e.g., Linux): Linux
- How you installed Ignite (`conda`, `pip`, source): source
- Python version: 3.8.6
- Any other relevant information:
    mypy version: mypy-0.901-cp38-cp38-manylinux2010_x86_64

--- a/ignite/handlers/checkpoint.py
+++ b/ignite/handlers/checkpoint.py
@@ -6,7 +6,7 @@ import tempfile
 import warnings
 from abc import ABCMeta, abstractmethod
 from collections import OrderedDict
-from tempfile import _TemporaryFileWrapper  # type: ignore[attr-defined]
+from tempfile import _TemporaryFileWrapper
 from typing import Any, Callable, Dict, List, Mapping, NamedTuple, Optional, Tuple, Union
 
 import torch
@@ -692,10 +692,10 @@ class DiskSaver(BaseSaveHandler):
         else:
             tmp_file = None
             tmp_name = ""
-            tmp = None  # type: _TemporaryFileWrapper
+            tmp = None 
             if rank == 0:
                 tmp = tempfile.NamedTemporaryFile(delete=False, dir=self.dirname)
-                tmp_file = tmp.file
+                tmp_file = tmp.file # type: ignore
                 tmp_name = tmp.name
             try:
                 func(checkpoint, tmp_file, **self.kwargs)

--- a/ignite/handlers/checkpoint.py
+++ b/ignite/handlers/checkpoint.py
@@ -6,7 +6,7 @@ import tempfile
 import warnings
 from abc import ABCMeta, abstractmethod
 from collections import OrderedDict
-from typing import Any, Callable, Dict, IO, List, Mapping, NamedTuple, Optional, Tuple, Union
+from typing import IO, Any, Callable, Dict, List, Mapping, NamedTuple, Optional, Tuple, Union
 
 import torch
 import torch.nn as nn
@@ -691,7 +691,7 @@ class DiskSaver(BaseSaveHandler):
         else:
             tmp_file = None
             tmp_name = ""
-            tmp : Optional[IO[bytes]] = None
+            tmp: Optional[IO[bytes]] = None
             if rank == 0:
                 tmp = tempfile.NamedTemporaryFile(delete=False, dir=self.dirname)
                 tmp_file = tmp.file  # type: ignore

--- a/ignite/handlers/checkpoint.py
+++ b/ignite/handlers/checkpoint.py
@@ -6,8 +6,7 @@ import tempfile
 import warnings
 from abc import ABCMeta, abstractmethod
 from collections import OrderedDict
-from tempfile import _TemporaryFileWrapper
-from typing import Any, Callable, Dict, List, Mapping, NamedTuple, Optional, Tuple, Union
+from typing import Any, Callable, Dict, IO, List, Mapping, NamedTuple, Optional, Tuple, Union
 
 import torch
 import torch.nn as nn
@@ -692,10 +691,10 @@ class DiskSaver(BaseSaveHandler):
         else:
             tmp_file = None
             tmp_name = ""
-            tmp = None 
+            tmp : Optional[IO[bytes]] = None
             if rank == 0:
                 tmp = tempfile.NamedTemporaryFile(delete=False, dir=self.dirname)
-                tmp_file = tmp.file # type: ignore
+                tmp_file = tmp.file  # type: ignore
                 tmp_name = tmp.name
             try:
                 func(checkpoint, tmp_file, **self.kwargs)

--- a/ignite/utils.py
+++ b/ignite/utils.py
@@ -158,7 +158,7 @@ def setup_logger(
         Added ``reset`` parameter.
     """
     # check if the logger already exists
-    existing = name is None or name in logging.root.manager.loggerDict  # type: ignore
+    existing = name is None or name in logging.root.manager.loggerDict
 
     # if existing, get the logger otherwise create a new one
     logger = logging.getLogger(name)

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -26,5 +26,4 @@ nltk
 pandas
 gym
 mkl
-#mypy
-mypy==0.812
+mypy

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -26,4 +26,3 @@ nltk
 pandas
 gym
 mkl
-mypy

